ruma-events-macros/src/event_enum.rs

```diff
@@ -575,17 +568,109 @@ fn field_return_type(name: &str) -> TokenStream {
 
 /// Custom keywords for the `event_enum!` macro
 mod kw {
-    syn::custom_keyword!(name);
+    syn::custom_keyword!(kind);
     syn::custom_keyword!(events);
 }
 
+pub enum EnumKind {
+    Basic(Ident),
+    Ephemeral(Ident),
+    Message(Ident),
+    State(Ident),
+    ToDevice(Ident),
+}
+
+impl EnumKind {
+    /// Return the span of this ident.
+    fn span(&self) -> Span {
+        match &self {
+            Self::Basic(i)
+            | Self::Ephemeral(i)
+            | Self::Message(i)
+            | Self::State(i)
+            | Self::ToDevice(i) => i.span(),
+        }
```

Comment:
These functions that just do stuff on the `Ident` could benefit from a `get_ident` helper fn.

Reply:
Beautiful! good thinking!